### PR TITLE
Shortener

### PR DESCRIPTION
**Pull request by bigblind**
Sorry, I forgot to switch to development instead of master yesterday.

### CODE REVIEW

File: saulify/cache.py [hunk 1]
```diff
@@ -66,15 +67,18 @@ def _make_key(namespace, fname, *args, **kwargs):
...
-
-def cached_function(namespace, expires=None):
+def cached_function(namespace, expires=None, serializer=_noop , deserializer=_noop):
     """This function is intended to be used as a decorator, for expensive operations.
 
     It first checks if the function has been called before during the request.
     If it hasn't it checks whether there's a value stored in Redis.
     If there isn't, the function is called.
 
+    Values are alwas saved as strings. To make sure you store all the information you want, and get it back out again,
```
Comment (iwasrobbed) [inline saulify/cache.py]: Nitpick: `always` spelling

File: saulify/common.py [hunk 2]
```diff
@@ -7,6 +7,8 @@
 from functools import wraps
 from flask.ext.login import current_user
 
+ALPHABET = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
```
Comment (iwasrobbed) [inline saulify/common.py]: Is this a remnant of the old slug generation code?

File: saulify/models.py [hunk 3]
```diff
@@ -33,3 +36,65 @@ def get_id(self):
...
+        db.session.commit()
+
+    # These methods don't take a 'self' argument, because they're used during class definition
+    def _serialize(obj):
+        return cache.serialize_expression({
+            'id': obj.id,
+            'updated': time.mktime(obj.updated.timetuple())+1e-6*obj.updated.microsecond, #converts datetime to timestamp
+            'url': obj.url,
+            'authors': obj.authors,
+            'markdown': obj.markdown,
```
Comment (iwasrobbed) [inline saulify/models.py]: Are there any concerns about data type size here for Markdown storage? That gives us roughly 64kb or maybe less for UTF-8 encodings

File: $1 [hunk 4]
```diff
@@ -0,0 +1 @@
+/usr/local/foreman/lib/foreman/cli.rb:91: warning: Insecure world writable dir /Users/frederikcreemers/dev/google-cloud-sdk in PATH, mode 040777
```
Comment (iwasrobbed) [inline $1]: I think this one slipped into the commit :)

**Comment by iwasrobbed** [comment]
Really nice work as usual, @bigblind! Just a few minor comments for you

**Comment by iwasrobbed** [comment]
As a side note, anything we need to update in the README file for setting this up locally?

**Comment by bigblind** [comment]
Only TEST_REDIS_URL for the cache.

File: saulify/models.py (hunk 3 above)
Comment (bigblind) [inline saulify/models.py]: For Postgres, there's no limit enforced by the system http://www.postgresql.org/docs/9.1/static/datatype-character.html

String values in Redis have a maximum size of 512 megabytes.
http://redis.io/topics/data-types
I don't think that will be a problem, but maybe we should have a check for it somewhere. I'm not sure what kind of error the python Redis library would raise in such a case.

File: $1 (hunk 4 above)
Comment (bigblind) [inline $1]: Yes it did. I've removed it now.

File: saulify/models.py (hunk 3 above)
Comment (iwasrobbed) [inline saulify/models.py]: Okay, no worries then. We'll just keep an eye out for any limits being hit in error reporting; seems like a fairly large limit though

**Pull request merged**